cpp/docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,453 @@
+
+## Doxygen
+
+The [doxygen tool](http://www.doxygen.nl/manual/index.html) is used to generate html pages from the C++ comments in the source code. Doxygen will recognize and parse block comments as well as perform specialized output formatting when encountering [doxgen commands](http://www.doxygen.nl/manual/commands.html).
+
+There are almost 200 commands (also called tags in this document) that doxygen recognizes in comment blocks. This document will provide guidance on which commands/tags to use and how to use them in the libcudf C++ source code.
+
+The doxygen process can be customized using the many, many configuration options in the [Doxyfile](../doxygen/Doxyfile).
+
+Here are some of the custom options in the Doxyfile for libcudf.
+| Option | Setting | Description |
+| ------ | ------- | ----------- |
+| PROJECT_NAME | libcudf | Title used on the main page |
+| PROJECT_NUMBER | 0.14 | Version number |
+| EXTENSION_MAPPING | cu=C++ cuh=C++ | Process `cu` and `cuh` as C++ |
+| INPUT | main_page.md regex.md unicode.md ../include | Embedded markdown files and source code directories to process |
+| FILE_PATTERNS | *.cpp *.hpp *.h *.c *.cu *.cuh | File extensions to process |
+
+## Block Comments
+
+The following block comment style should be used for all C++ doxygen comments.
+
+```c++
+/**
+ * description text and
+ * doxygen tags go here
+ */
+```
+
+The block should start with `/**` and end with `*/` only and with nothing else on those lines.
+(e.g. do not add dashes `-----` or extra asterisks `*****` in these lines).
+The block must be placed immediately before the source code line in which it is referring.
+The block may be indented to line up vertically with the item they are documenting as appropriate. See the [Example](#the-example) section below.
+
+Each line in the comment block between the `/**` and `*/` lines should start with a space followed by an asterisk. Any text on these lines, including tag declarations, should start after a single space after the asterisk.
+
+## Tag/Command names
+
+Use `@` to prefix doxygen commands (e.g. `@brief`, `@code`, etc.)
+
+## Markdown
+
+The doxygen tool supports a limited set of markdown format in the comment block including links, tables, lists, etc.
+
+In some cases a trade-off may be required for readability in the source text file versus the readability in the doxygen formatted web pages.
+
+For example, there are some limitations on readability with '%' character and pipe character '|' within a markdown table.
+
+Also, try to avoid using direct hTML tags. Although doxygen supports markdown and markdown supports HTML tags, the HTML support for doxygen's markdown is also limited.
+
+## The Example
+
+The following example will cover most of the doxygen block comment and tag styles
+for documenting C++ code in libcudf.
+
+```c++
+/**
+ * @file source_file.cpp
+ * @brief Description of source file contents
+ *
+ * Longer description of the source file contents.
+ */
+
```

Comment:
Maybe add a group definition example here?

Reply:
The `@ingroup` example is included in the comment below.